frontend/public/components/utils/triggers.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+
+import { K8sResourceKind } from '../../module/k8s';
+import { ResourceLink } from './';
+
+export const Triggers: React.SFC<TriggersProps> = ({ resource }) => {
+  const triggers = _.get(resource, 'spec.triggers');
+  const getTriggerRefType = trigger => {
```

Comment:
This looks like it can be simplified to:
```javascript
trigger => validTriggers.has(trigger) ? trigger.toLowerCase() : null
```

Reply:
@spadgett and I discussed a similar approach, but opted for the switch so the code was explicit.  It's possible you can convince us otherwise...

Reply:
Since there is no unique behavior in any of the cases, I don't see a reason to be explicit here (YAGNI). Making an `enum` with each trigger type would be better, as well.

Reply:
Don't have a strong opinion on this as long as we have `validTriggers`. Agree on the enum